perf/perf_demangle.py

```diff
@@ -0,0 +1,158 @@
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def run_cmd(self, cmd):
+        """
+        run command on SUT as root
+        """
+        result = process.run(cmd, shell=True, ignore_status=True)
+        output = result.stdout_text + result.stderr_text
+        return output
+
+    def install_google_benchmark(self):
+        self.run_cmd("git clone https://github.com/google/benchmark.git")
+        os.chdir("benchmark")
+        self.run_cmd("git clone https://github.com/google/googletest.git")
+        os.makedirs("build", exist_ok=True)
+        os.chdir("build")
+        self.run_cmd("cmake -DCMAKE_BUILD_TYPE=Release -DBENCHMARK_DOWNLOAD_DEPENDENCIES=ON ..")
+        self.run_cmd(f"make -j$(nproc)")
+        self.run_cmd("sudo make install")
+        os.chdir("../../")
+
+    def write_test_cpp(self):
+        cpp_code = """
+#include <benchmark/benchmark.h>
+
+static __attribute__ ((noinline)) int my_really_big_function()
+{
+    for(size_t i = 0; i < 1000; ++i)
+    {
+        benchmark::DoNotOptimize(i % 5);
+    }
+    return 0;
+}
+
+static __attribute__ ((noinline)) void caller1()
+{
+    for(size_t i = 0; i < 1000; ++i)
+    {
+        benchmark::DoNotOptimize(my_really_big_function());
+        benchmark::DoNotOptimize(i % 5);
+    }
+}
+
+static __attribute__ ((noinline)) void myfun(benchmark::State& state)
+{
+    while(state.KeepRunning())
+    {
+        caller1();
+    }
+}
+
+BENCHMARK(myfun);
+BENCHMARK_MAIN();
+"""
+        with open("main.cpp", "w") as f:
+            f.write(cpp_code)
+
+    def compile_benchmark(self):
+        self.run_cmd("clang++ main.cpp -o main -fno-omit-frame-pointer -O0 -lpthread -lbenchmark")
+
+    def record_perf(self):
+        self.run_cmd("perf record -g -o perf_output.data ./main")
+
+    def get_perf_report(self):
+        return self.run_cmd("perf report --sort dso,symbol --no-children -i perf_output.data --stdio").splitlines()
+
+    def is_mangled(self, symbol):
+        return re.match(r'^_Z[\w\d_]*$', symbol)
+
+    def analyze_symbols(self, symbols):
+        mangled = sum(1 for s in symbols if self.is_mangled(s))
+        total = len(symbols)
+        demangled = total - mangled
+        return (
+            (mangled / total) * 100 if total else 0,
+            (demangled / total) * 100 if total else 0
+        )
+
+    def parse_report_and_test(self, lines):
+        symbols = []
+        for line in lines:
+            match = re.search(r'\[\.\]\s+(.+)', line)
+            if match:
+                symbols.append(match.group(1).strip())
+
+        mangled_pct, demangled_pct = self.analyze_symbols(symbols)
+        print(f"\n[*] Total symbols: {len(symbols)}")
+        print(f"[+] Mangled: {mangled_pct:.2f}%, Demangled: {demangled_pct:.2f}%")
+
+        self.assertGreaterEqual(demangled_pct, 35, "[FAIL] Symbols appear to be mostly mangled.")
+
+    def cleanup(self):
+        for path in ["benchmark", "main.cpp", "main", "perf.data"]:
```

Comment:
you are using perf_output.data then cleanup should also have perf_output.data not perf.data